deps/jerry/jerry-ext/handle-scope/handle-scope-allocator.c

```diff
@@ -0,0 +1,196 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include "handle-scope-internal.h"
+
+static jerryx_handle_scope_t kJerryXHandleScopeRoot = {
+  .handle_count = 0,
+  .handle_ptr = NULL,
+};
+static jerryx_handle_scope_t *kJerryXHandleScopeCurrent = &kJerryXHandleScopeRoot;
+static jerryx_handle_scope_pool_t kJerryXHandleScopePool = {
+  .count = 0,
+  .start = NULL,
+  .end = NULL,
+};
+
+#define kJerryXHandleScopePoolPrelistLast \
+  kJerryXHandleScopePool.prelist + JERRY_X_HANDLE_SCOPE_PRELIST_SCOPE_COUNT - 1
+
+#define JerryXHandleScopePrelistIdx(scope) (scope - kJerryXHandleScopePool.prelist)
+
+
+/**
+ * Get current handle scope top of stack.
+ */
+jerryx_handle_scope_t *
+jerryx_handle_scope_get_current (void)
+{
+  return kJerryXHandleScopeCurrent;
+}
+
+
+/**
+ * Get root handle scope.
+ */
+jerryx_handle_scope_t *
+jerryx_handle_scope_get_root (void)
+{
+  return &kJerryXHandleScopeRoot;
+}
```

Comment:
`jerryx_handle_scope_get_current` and `jerryx_handle_scope_get_root` could be static and inlined.